src/torchcodec/decoders/_video_decoder.py

```diff
@@ -476,10 +476,12 @@ def _read_custom_frame_mappings(
         )
 
     frame_data = [
-        (float(frame[pts_key]), frame["key_frame"], float(frame[duration_key]))
+        (int(frame[pts_key]), frame["key_frame"], int(frame[duration_key]))
         for frame in input_data["frames"]
     ]
-    all_frames, is_key_frame, duration = map(torch.tensor, zip(*frame_data))
+    all_frames = torch.tensor([x[0] for x in frame_data], dtype=torch.int64)
+    is_key_frame = torch.tensor([x[1] for x in frame_data], dtype=torch.bool)
+    duration = torch.tensor([x[2] for x in frame_data], dtype=torch.int64)
```

Comment:
At this point, it's not worth creating `frame_data` anymore, since the `x[...]` indexing becomes un-pythonic. I think it'd be preferable to just define `all_frames` and the other tensors as e.g.:

```py
all_frames = torch.tensor([int(frame[pts_key]) for frame in input_data["frames"]], dtype=torch.int64)
...
```